internal/armadactl/scheduling_control.go

```diff
@@ -0,0 +1,119 @@
+package armadactl
+
+import (
+	"fmt"
+
+	"github.com/armadaproject/armada/internal/common/slices"
+	"github.com/armadaproject/armada/pkg/api"
+	"github.com/armadaproject/armada/pkg/client/queue"
+)
+
+func (a *App) cordonQueue(q *api.Queue) error {
+	q.SchedulingPaused = true
+	newQueue, err := queue.NewQueue(q)
+	if err := a.Params.QueueAPI.Update(newQueue); err != nil {
```

Comment:
I'd argue it would be better here to have a cordon rpc call that just takes the queue name. That way you know that the only thing that can be updated by this function is the cordon flag.  As it stands a user could inadvertantly update e.g. queuePriority by calling this function, which I'd argue would be somewhat uinexpected!